src/host.rs

```diff
@@ -68,13 +106,19 @@ impl Udp {
         }
     }
 
+    fn is_port_assigned(&self, port: u16) -> bool {
+        self.binds.keys().any(|a| a.port() == port)
+    }
+
     pub(crate) fn bind(&mut self, addr: SocketAddr) -> io::Result<UdpSocket> {
         let (tx, rx) = mpsc::unbounded_channel();
 
         if self.binds.insert(addr, tx).is_some() {
             return Err(io::Error::new(io::ErrorKind::AddrInUse, addr.to_string()));
         }
 
+        trace!(?addr, protocol = %"UDP", "Bind");
```

Comment:
suggestion to avoid formatting a string with display, but I think this is better done as a span.
```suggestion
        trace!(?addr, protocol = "UDP", "Bind");
```

Reply:
I was going for consistency, `protocol = UDP` when done via `Display` vs. `protocol = "UDP"` when done manually.